server/src/main/java/io/crate/replication/logical/metadata/pgcatalog/PgSubscriptionRelTable.java

```diff
@@ -44,38 +41,37 @@
 
     public static SystemTable<PgSubscriptionRelTable.PgSubscriptionRelRow> create() {
         return SystemTable.<PgSubscriptionRelTable.PgSubscriptionRelRow>builder(IDENT)
-            .add("srsubid", INTEGER, r -> r.subOid())
-            .add("srrelid", REGCLASS, r -> r.relOid())
-            .add("srsubstate", STRING, ignored -> null)
+            .add("srsubid", INTEGER, PgSubscriptionRelRow::subOid)
+            .add("srrelid", REGCLASS, PgSubscriptionRelRow::relOid)
+            .add("srsubstate", STRING, PgSubscriptionRelRow::state)
+            .add("srsubstate_reason", STRING, PgSubscriptionRelRow::state_reason)
+            // CrateDB doesn't have Log Sequence Number per table, only a seqNo per shard (see SysShardsTable)
+            .add("srsublsn", LONG, ignored -> null)
             .build();
     }
 
-    public static Iterable<PgSubscriptionRelTable.PgSubscriptionRelRow> rows(LogicalReplicationService logicalReplicationService,
-                                                                             Schemas schemas) {
+    public static Iterable<PgSubscriptionRelTable.PgSubscriptionRelRow> rows(LogicalReplicationService logicalReplicationService) {
         return () -> {
             Stream<PgSubscriptionRelTable.PgSubscriptionRelRow> s = logicalReplicationService.subscriptions().entrySet().stream()
                 .mapMulti(
                     (e, c) -> {
                         var sub = e.getValue();
-                        InformationSchemaIterables.tablesStream(schemas)
-                            .filter(t -> {
-                                if (t instanceof DocTableInfo dt) {
-                                    return e.getKey().equals(REPLICATION_SUBSCRIPTION_NAME.get(dt.parameters()));
-                                }
-                                return false;
-                            })
-                            .forEach(t -> c.accept(
+                        sub.relations().forEach(
+                            (r, rs) -> c.accept(
```

Comment:
Could use  `flatMap` + `map` here, but that's mostly personal preference because it's the pattern I'm used to read. Fine to keep it as is.